### ISSUE TITLE
Proxy-Cache for official images isn't working

### ISSUE THREAD

**Issue by fandujar** [goharbor/harbor#13733]
**Expected behavior and actual behavior:**
Commands like `docker pull myharbor.mycompany.com/myproxycache/alpine:latest` would result in image pulled 

**Steps to reproduce the problem:**

- Create a project with proxy-cache enabled linked to dockerhub endpoint (i used dockerhub as my project name)
- Use docker client to pull a official image like alpine or nginx

**Versions:**
Please specify the versions of following systems.

- harbor version: [2.1.1]
- docker engine version: [19.03.13 (Community)]

Result from docker pull command:

`docker pull myharbor.mycompany.com/dockerhub/alpine:latest
Error response from daemon: unknown: repository dockerhub/alpine not found`

**Comment by EdKingscote** [comment]
Have you tried `docker pull myharbor.mycompany.com/dockerhub/library/alpine:latest`  there is a footnote at the bottom of the docs at https://goharbor.io/docs/2.1.0/administration/configure-proxy-cache/#how-harbor-proxy-cache-works about single level repositories and official images.

**Comment by alfsch** [comment]
I have the same errors with exact the same setup. When I use

` docker pull myharbor.mycompany.comdockerhub/library/alpine:latest`

I get a different error
`
Error response from daemon: missing or empty Content-Type header`

When I use the command like in the bug description I get the same error as in issyu description.

**Comment by fabianotessarolo** [comment]
Yeah, indeed, that worked here @EdKingscote @fandujar.

A related issue here, but for Transparent proxy is: 
Problem: https://github.com/goharbor/harbor/issues/8082
Feature Request: https://github.com/goharbor/harbor/issues/13579

Seems like [NGINX rewrite suggestion for Transparent Proxy](https://github.com/goharbor/harbor/issues/8082#issuecomment-731959494), should take into consideration this,

**Comment by fandujar** [comment]
@EdKingscote I tried, but got same error @alfsch 

`Error response from daemon: missing or empty Content-Type header`

**Comment by alfsch** [comment]
the problem is docker 20.10 see #13740

**Comment by EdKingscote** [comment]
I ran in to the Content-Type header problem last night as well, but only on machines that all just happened to be internal, so I figured it was a quirk of HAProxy being our front-end before I had to call it a night. Thanks @alfsch for sharing the link to that new issue, as it turns out purely by chance all of our external hosts are using older docker engines than our internal ones - that saved me a good chunk of investigation time this morning! Hopefully it won't take too long for a fix to materialize somewhere.
